sds_data_manager/lambda_code/SDSCode/pipeline_lambdas/batch_starter.py

```diff
@@ -681,6 +681,57 @@ def calculate_pointing_date_range(session, pointing_id):
     return start_date, end_date
 
 
+def calculate_repoint_table_date_range(session, file_obj):
+    """Calculate the date range for a repoint-table.
+
+    The end date can easily be gotten from the filename. In order to determine
+    the start date, we query the database and use the end date from the previous
+    repoint table.
+
+    Notes
+    -----
+    Repoint file is used to kick off the pointing_attitude job only.
+    This date range is used to query attitude kernel file(s). If
+    other jobs become dependent on triggering off of the repoint file,
+    please revisit this logic.
+
+
+    Parameters
+    ----------
+    session : sqlalchemy.orm.Session
+        Database session.
+    file_obj : SPICEFilePath
+        Repoint table file object.
+
+    Returns
+    -------
+    tuple
+        A tuple containing the start date and end date in the format YYYYMMDD.
+    """
+    # Query the pointing table to find the pointing information.
+    end_date = file_obj.spice_metadata["end_date"]
+    previous_entries = (
+        session.query(models.RepointFiles)
+        .filter(models.RepointFiles.end_date <= end_date)
+        .order_by(models.RepointFiles.end_date, models.RepointFiles.version)
+        .all()
```

Comment:
Can you descend the ordering and then get the first entry instead? Then update the logic below.
```suggestion
        .order_by(models.RepointFiles.end_date.desc(), models.RepointFiles.version.desc())
        .first()
```

Reply:
Just realized that this is already in "dev". Was reviewing something that already went in 😂

Reply:
My sqlalchemy is pretty weak. It looks like I should have ordered descending and then used `.limit(2)` to just get the first two results. I can write a ticket for that change.